app/src/main/java/eu/kanade/tachiyomi/data/download/DownloadCache.kt

```diff
@@ -75,12 +75,14 @@ class DownloadCache(
 
         checkRenew()
 
-        val files = mangaFiles[manga.id]?.toSet() ?: return false
-        return files.any { file ->
-            provider.getValidChapterDirNames(chapter).any {
-                it.toLowerCase() == file.toLowerCase()
+        val files = mangaFiles[manga.id]?.toHashSet() ?: return false
+        val validChapterDirNames = provider.getValidChapterDirNames(chapter)
+        validChapterDirNames.forEach {
```

Comment:
```kotlin
return provider.getValidChapterDirNames.any { it in files }
```

?

Reply:
I'll double check tomorrow if that keeps same performance

Reply:
Cool looked it up. In calls contains behind the scenes. So should be able to rewrite it that way